src/index.js

```diff
@@ -0,0 +1,266 @@
+  }
+};
+
+const HandleCityNameError = () =>{
+  const cityErrorMessageModal = document.getElementById('cityErrorDialog');
+  cityErrorMessageModal.showModal();
+  let errorData = {
+    name: '??',
+    weather: '??',
+    weatherIconCode: '??',
+    weatherDescription: '??',
+    realTempValue: '??'
+  };
+  updateState(errorData);
+  document.getElementById('realTempValue').textContent = errorData.realTempValue;
+  changeRealTempValueColor();
+  changeRealWeatherDetails();
+}
+
+const closeErrorModal = () =>{
+  const cityErrorMessageModal = document.getElementById('cityErrorDialog');
+  cityErrorMessageModal.close();
+};
+
+//Wave 2
+const changeGardenTempValueColorAndLandscape = () =>{
+  let data = {};
+  for (const idx in tempProperties){
+    let temp = tempProperties[idx].temp;
+    if (temp === 49 && state.gardenTempValue<= temp){
+      data.gardenTempValueColor = tempProperties[idx].color;
+      data.landscape = tempProperties[idx].landscape;
+      break;
+    }else if (state.gardenTempValue >= temp){
+      data.gardenTempValueColor = tempProperties[idx].color;
+      data.landscape = tempProperties[idx].landscape;
+      break;
+    }
+  }
+  document.getElementById('tempValue').classList.remove(state.gardenTempValueColor)
+  updateState(data);
+  document.getElementById('tempValue').classList.add(state.gardenTempValueColor);
+  document.getElementById('landscape'). textContent = state.landscape;
+};
+
+
+
+const changeRealTempValueColor = () =>{
+  let colorData ={};
+  for (const idx in tempProperties){
+    let temp = tempProperties[idx].temp;
+    if (temp === 49 && state.realTempValue<= temp){
+      colorData.realTempValueColor = tempProperties[idx].color;
+      break;
+    }else if (state.realTempValue >= temp){
+      colorData.realTempValueColor = tempProperties[idx].color;
+      break;
+    }
+  }
+  document.getElementById('realTempValue').classList.remove(state.realTempValueColor)
+  updateState(colorData);
+  document.getElementById('realTempValue').classList.add(state.realTempValueColor)
+
+};
+
+const changeRealWeatherDetails = () =>{
+  const weatherDetails = document.getElementById('weather-details');
+  weatherDetails.textContent = state.weatherDescription[0].toUpperCase() + state.weatherDescription.substring(1);
+
+  const weatherIcon = document.getElementById('weather-icon');
+  if (state.weatherIconCode != '??'){
+    weatherIcon.src = `https://openweathermap.org/img/wn/${state.weatherIconCode}@2x.png`;
+  } else {
+    weatherIcon.src = 'assets/unknown-weather-icon.png'
+  }
+  };
+
+const increaseTemp = () =>{
+  state.gardenTempValue += 1;
+  const temp = document.getElementById('tempValue');
+  temp.textContent = state.gardenTempValue;
+  changeGardenTempValueColorAndLandscape();
+};
+
+const decreaseTemp = () =>{
+  state.gardenTempValue -= 1;
+  const temp = document.getElementById('tempValue');
+  temp.textContent  = state.gardenTempValue;
+  changeGardenTempValueColorAndLandscape();
+};
```

Comment:
Since `increaseTemp` and `decreaseTemp` are very similar, another approach we could take is to combine them into 1 function:
```javascript
const setTemp = (changeInTemp) => {
  state.gardenTempValue += changeInTemp;
  const temp = document.getElementById('tempValue');
  temp.textContent = state.gardenTempValue;
  changeGardenTempValueColorAndLandscape();
};
```
Then where we register the handlers, we could call our new function with a parameter by doing so inside an anonymous function:
```javascript
  increaseTempButton.addEventListener('click', () => { setTemp(1) });
  decreaseTempButton.addEventListener('click', () => { setTemp(-1) });
```